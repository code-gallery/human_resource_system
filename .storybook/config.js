import { configure } from '@storybook/react'

const loadStories = () => {
  require('../src/stories')
}

configure(loadStories, module)
