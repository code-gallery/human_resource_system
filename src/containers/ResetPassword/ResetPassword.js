import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import Notifications, { notify } from 'react-notify-toast'
import { NOTIFICATION_TIMEOUT, ROUTE_URL, getApiUrl } from 'containers/constants'
import Form from 'components/Abstract/Form'
import httpFetch from 'utils/httpFetch'
import 'containers/Register/style.css'

class ResetPassword extends Form {
  constructor(props) {
    super(props)
    this.state = {
      password: '',
      confirmPassword: ''
    }

    this.genericError = 'An error occured. Please try again.'
    this.onSubmit = this.onSubmit.bind(this)
  }

  componentDidMount() {
    document.body.classList.add('login-page')
  }

  componentWillUnmount() {
    document.body.classList.remove('login-page')
  }

  onSubmit(event) {
    event.preventDefault()
    const { password, confirmPassword } = this.state
    const { token } = this.props.match.params

    if (password !== '' && confirmPassword !== '' && password === confirmPassword) {
      httpFetch(getApiUrl('resetPassword'), {
        method: 'POST',
        body: JSON.stringify({
          token,
          password
        })
      })
        .then((data) => {
          if (data.status === 'success') {
            notify.show(data.message, 'success', NOTIFICATION_TIMEOUT)
          } else {
            const msg = (data.errors) ? data.errors[0].message : this.genericError
            notify.show(msg, 'error', NOTIFICATION_TIMEOUT)
          }
        })
        .catch(() => {
          notify.show(this.genericError, 'error', NOTIFICATION_TIMEOUT)
        })
    } else {
      notify.show('Please enter matching passwords', 'error', NOTIFICATION_TIMEOUT)
    }
    this.setState({
      password: '',
      confirmPassword: ''
    })
  }

  render() {
    const { password, confirmPassword } = this.state
    return (
      <form className="Login-page">
        <Notifications />
        <section className="Login-block">
          <h1 className="Login-logo">
            {/* external link - not react route - leave <a> */}
            <a href={ROUTE_URL.publicSiteHome}><span>APPII</span></a>
          </h1>
          <p className="Login-title">Reset your password</p>
          <div className="form-group">
            <input
              name="password"
              type="password"
              placeholder="Password"
              value={password}
              onChange={this.onChangeInput}
            />
          </div>
          <div className="form-group">
            <input
              name="confirmPassword"
              type="password"
              placeholder="Confirm password"
              value={confirmPassword}
              onChange={this.onChangeInput}
            />
          </div>
          <div className="form-group">
            <button className="border-btn" onClick={this.onSubmit}>
              Reset
            </button>
          </div>
          <div className="form-group">
            <p className="form-devider">or</p>
          </div>
          <div className="form-group">
            <Link to={ROUTE_URL.login} className="border-btn">
              Login
            </Link>
          </div>
        </section>
      </form>
    )
  }
}

ResetPassword.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.object.isRequired
  })
}

export default ResetPassword
