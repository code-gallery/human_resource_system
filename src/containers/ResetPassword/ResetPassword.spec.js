import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import config from 'store'
import ResetPassword from './ResetPassword'

describe('<ResetPassword />', () => {
  const props = {
    match: {
      params: {}
    }
  }

  it('renders correctly', () => {
    const { store } = config
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <ResetPassword {...props} />
        </Router>
      </Provider>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
