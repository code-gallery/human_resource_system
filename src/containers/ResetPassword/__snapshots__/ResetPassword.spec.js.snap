// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPassword /> renders correctly 1`] = `
<form
  className="Login-page"
>
  <div
    id="notification-wrapper"
  />
  <section
    className="Login-block"
  >
    <h1
      className="Login-logo"
    >
      <a
        href="https://appii.io"
      >
        <span>
          APPII
        </span>
      </a>
    </h1>
    <p
      className="Login-title"
    >
      Reset your password
    </p>
    <div
      className="form-group"
    >
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        name="confirmPassword"
        onChange={[Function]}
        placeholder="Confirm password"
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <button
        className="border-btn"
        onClick={[Function]}
      >
        Reset
      </button>
    </div>
    <div
      className="form-group"
    >
      <p
        className="form-devider"
      >
        or
      </p>
    </div>
    <div
      className="form-group"
    >
      <a
        className="border-btn"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </section>
</form>
`;
